src/test/mir-opt/const_prop/const_prop_fails_gracefully.rs

```diff
@@ -23,8 +23,8 @@ fn main() {
 // START rustc.main.ConstProp.after.mir
 //  bb0: {
 //      ...
-//      _4 = const Scalar(AllocId(1).0x0) : &i32;
-//      _3 = const Scalar(AllocId(1).0x0) : &i32;
+//      _4 = const {pointer} : &i32;
+//      _3 = const {pointer} : &i32;
 //      _2 = const Value(Scalar(AllocId(1).0x0)) : *const i32;
```

Comment:
This is weirdly inconsistent.

Also @oli-obk @wesleywiser could you check these `-Zdump-mir` changes. This is a regression in terms of information content but I am not sure if anyone cares.

Reply:
Mir dumps should probably dump the content of referenced allocations, like it dumps promoteds. Without showing alloc IDs this would be impossible to figure out though. I think mir dumps should keep dumping alloc ids

Reply:
This is where I got stuck. I still haven't been able to chase down where this Debug print comes from. Maybe we should open another issue?

Reply:
You are stuck trying to find out where the old printing for references came from, or the new one, or the one for raw pointers?

Reply:
The one for raw pointers. But with @oli-obk's comment in mind, maybe the opposite fix is more appropriate, replacing the new pretty printed references with a more verbose format?

Reply:
Well the PR changes behavior in an undesirable way when printing references, that part is in scope.

What is being debug-printed here is a `ConstKind`, I think. You could try removing the `Debug` impl for that and see where compilation fails.

Reply:
Good idea, thanks!